scripts/zkai_native_attention_mlp_adapter_compression_ablation_gate.py

```
@@ -0,0 +1,712 @@
+    resolved = resolved_parent / candidate.name
+    evidence_root = EVIDENCE_DIR.resolve()
+    if evidence_root not in resolved.parents:
+        raise AdapterCompressionAblationError("output path must stay under docs/engineering/evidence")
+    if resolved.suffix != suffix:
+        raise AdapterCompressionAblationError(f"output path must end with {suffix}")
+    return resolved
+
+
+def write_text_atomic(path: pathlib.Path, text: str) -> None:
+    if path.exists() and path.is_symlink():
+        raise AdapterCompressionAblationError("refusing to write through symlink")
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)